src/audio/copier/copier.c

```diff
@@ -887,7 +887,7 @@ static int copier_copy(struct comp_dev *dev)
 		sink_c = buffer_acquire(sink);
 		sink_dev = sink_c->sink;
 		processed_data.sink_bytes = 0;
-		if (sink_dev->state == COMP_STATE_ACTIVE) {
+		if (sink_dev && sink_dev->state == COMP_STATE_ACTIVE) {
```

Comment:
@RanderWang this can never happen, right? During comp_disconnect, we have disabled interrupts. So it is not possible that the buffer is dsconnected from the sink dev but it hasn't been deleted from the list and freed.

Reply:
in different stream case, not in the same stream.

Reply:
I meant the buffer is not disconnected for the state check in comp_disconnect.  Ignore it since I remove the logic